### PR TITLE
Issue 798 combine tag input

### LINKED ISSUE #798: [TagInput] Combine TagInput/TagInputStateful
### Wanted 
Combine the “dumb” TagInput component with the TagInputStateful component from https://github.com/sociomantic-tsunami/nessie-ui/pull/752

### PR DESCRIPTION
**Pull request by conor-cafferkey-sociomantic**
Closes #798:
- Combines TagInput and TagInputStateful into a single TagInput
- Simplifies Tag and TagInput components
- **Temporarily disabled TagInput driver tests**

### CODE REVIEW

File: src/TagInput/index.jsx [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+        this.setState( ( { filteredOptions, tags } ) =>
+        {
+            const option = getOption( id, filteredOptions );
+            const newTags = [ ...tags, option.text ];
+
+            const { onChange } = this.props;
+            if ( typeof onChange === 'function' )
+            {
+                onChange( { tags: newTags } );
+            }
```
Comment (damian-rodriguez-sociomantic) [inline src/TagInput/index.jsx]: this should be also added to `handleClickClose` and `handleKeyDown`

File: src/TagInput/index.jsx [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+                    onMouseOutOption  = { this.handleMouseOutOption }
+                    onMouseOverOption = { this.handleMouseOverOption }
+                    options           = { listBoxOptions } />
+            </ScrollBox>
+        );
+
+
+        return (
+            <TagInputwithDropdown
+                { ...this.props }
```
Comment (damian-rodriguez-sociomantic) [inline src/TagInput/index.jsx]: `onChange` event should be excluded from this in case is passed as a prop.

File: src/TagInput/index.jsx (hunk 1 above)
Comment (conor-cafferkey-sociomantic) [reply src/TagInput/index.jsx]: Thanks for catching!

**Comment by daniel-martic-sociomantic** [comment]
Not just working, but also looking good 👍
Minor suggestion: at the moment, if we pass some values through `tags` prop, then these tags are basically read only and, once rendered, cannot be changed/removed; input also becomes read only. In that case, it would be good if we have a way of passing some default ("starting") tags to input that can later be removed or kept, while adding other/new tags to the input.

**Pull request merged**